s3path.py

```diff
@@ -898,15 +910,15 @@ def is_file(self):
         except ClientError:
             return False
 
-    def iterdir(self):
+    def iterdir(self: Self) -> Generator[Self, None, None]:
         """
         When the path points to a Bucket or a key prefix, yield path objects of the directory contents
         """
         self._absolute_path_validation()
         for name in self._accessor.listdir(self):
             yield self._make_child_relpath(name)
 
-    def glob(self, pattern):
+    def glob(self: Self, pattern: str) -> Generator[Self, None, None]:
```

Comment:
same comment here on the self type hint
You get me :-)
I wont write it for all the rest of the places